crates/radix-engine-toolkit/src/transaction_types/traverser/auxiliary/reserved_instructions.rs

```diff
@@ -39,36 +39,66 @@ impl ReservedInstructionsDetector {
     fn on_instruction(&mut self, instruction: &InstructionV2, _: usize) {
-        let InstructionV2::CallMethod(CallMethod {
-            address,
-            method_name,
-            ..
-        }) = instruction
-        else {
-            return;
-        };
-
-        if is_account(address)
-            && contains!(
-                method_name => [
-                    ACCOUNT_LOCK_FEE_IDENT,
-                    ACCOUNT_LOCK_FEE_AND_WITHDRAW_IDENT,
-                    ACCOUNT_LOCK_FEE_AND_WITHDRAW_NON_FUNGIBLES_IDENT,
-                ]
-            )
-        {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccountLockFee);
-        } else if is_account(address) && method_name == ACCOUNT_SECURIFY_IDENT {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccountSecurify);
-        } else if is_identity(address) && method_name == IDENTITY_SECURIFY_IDENT
-        {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccountLockFee);
-        } else if is_access_controller(address) {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccessControllerMethod);
+        // TODO: Make use of the typed manifest invocations - they would make
+        // some of the logic in here easier.
+        match instruction {
+            InstructionV2::CallMethod(CallMethod {
+                address,
+                method_name,
+                ..
+            }) => {
+                if is_account(address)
+                    && contains!(
+                        method_name => [
+                            ACCOUNT_LOCK_FEE_IDENT,
+                            ACCOUNT_LOCK_FEE_AND_WITHDRAW_IDENT,
+                            ACCOUNT_LOCK_FEE_AND_WITHDRAW_NON_FUNGIBLES_IDENT,
+                        ]
+                    )
+                {
+                    self.reserved_instructions
+                        .insert(ReservedInstruction::AccountLockFee);
+                } else if is_account(address)
+                    && method_name == ACCOUNT_SECURIFY_IDENT
+                {
+                    self.reserved_instructions
+                        .insert(ReservedInstruction::AccountSecurify);
+                } else if is_identity(address)
+                    && method_name == IDENTITY_SECURIFY_IDENT
+                {
+                    self.reserved_instructions
+                        .insert(ReservedInstruction::AccountLockFee);
+                } else if is_access_controller(address) {
+                    self.reserved_instructions
+                        .insert(ReservedInstruction::AccessControllerMethod);
+                }
+            }
+            InstructionV2::CallMetadataMethod(CallMetadataMethod {
+                address,
+                method_name,
+                args,
+            }) => {
+                if method_name == METADATA_SET_IDENT {
```

Comment:
Sorry - hidden additional requirement - we should probably prevent locking of `owner_keys` too, as that would also cause issues with account administration

Reply:
(Although in general, we don't want to prevent locking of other pieces of metadata because that would prevent people updating their dApp definition accounts)